packages/beacon-node/src/execution/engine/types.ts

```diff
@@ -109,9 +113,17 @@ type ExecutionPayloadRpcWithBlockValue = {
 };
 type ExecutionPayloadResponse = ExecutionPayloadRpc | ExecutionPayloadRpcWithBlockValue;
 
-export type ExecutionPayloadBodyRpc = {transactions: DATA[]; withdrawals: WithdrawalV1[] | null};
+export type ExecutionPayloadBodyRpc = {
+  transactions: DATA[];
+  withdrawals: WithdrawalV1[] | null;
+  depositReceipts: DepositReceiptV1[] | null;
```

Comment:
we also need to type for undefined here as payloadbody might not have this pre 6110 (perhaps withdrawals too)

Reply:
> we also need to type for undefined here as payloadbody might not have this pre 6110 (perhaps withdrawals too)

I will need to double check on this. I don't know what the behaviour is like in the case of withdrawals pre-capella

Reply:
> yep should be typed as `T | null | undefined`, since the field many not be declared at all. `ExecutionPayloadBody` is okay to only have null to represent absence of that data

For reference, Besu's behaviour is that after 6110 is implemented, it will return execution payload with `depositReceipts` present in `getPayloadV6110`, `getPayloadV3`, `getPayloadV2` and `getPayloadV1`. For pre-6110, `depositReceipts` has null value. Post-6110 has a list or empty list. 
Similarly `withdrawal` is present will null value if you call `getPayloadV1` pre-shanghai

It is true that for users running older version of Besu this field will be absent. Not sure about other EL client.